index.js

```diff
@@ -79,8 +152,52 @@ function sparqlProxy (options) {
       }, options.timeout)
     }
 
-    return client[queryOperation](query, currentQueryOptions).then((result) => {
```

Comment:
We have the query promess

client[queryOperation](query, currentQueryOptions).then((result) => ...

That can be passed to a wrapper function with cache capabilities. 

`withCache(promise)` that returns exactly the same as `promise`

But inside it returns and sets the values when it applies.

Reply:
Take for example: https://github.com/PacktPublishing/Node.js-Design-Patterns-Third-Edition/blob/master/11-advanced-recipes/04-batching-and-caching/totalSalesCache.js that wraps another function with the same signature


This allows to keep the code less complex